### DISCUSSION TITLE
Feature to specify custom lexicon

### DISCUSSION THREAD

**Discussion by francois-vz** [coqui-ai/TTS#2770] [Ideas]
Hi everyone, 

Just a suggestion that it might be worthwhile to add a feature that allows the addition of custom phonetic words. 
I know that you can remap words for g2p in espeak such as https://www.youtube.com/watch?v=493xbPIQBSU, but I think some optional parameter where you can specify some TSV such as:

```
test    tɛst
```

would be useful to the community (especially those working on applications for which g2p is not optimal). It's not too hard to hardcode the substitution of certain words in the source code, but a cleaner approach sure would be nice.

**Comment by omkarade** [comment]
Yes, adding a feature to specify a custom lexicon in the form of a TSV (Tab-Separated Values) file, where users can map words to their corresponding phonetic representations, would be a valuable addition to the project. This feature would allow users to customize the grapheme-to-phoneme (g2p) conversion for specific words, making the system more versatile and applicable to a broader range of applications. It would be a cleaner and more user-friendly solution compared to hardcoding substitutions directly in the source code.
